Content.Client/Light/Components/HandheldLightComponent.cs

```diff
@@ -17,6 +20,37 @@ namespace Content.Client.Light.Components
     public sealed class HandheldLightComponent : SharedHandheldLightComponent, IItemStatus
     {
         public byte? Level;
+        public bool Activated;
+
+        /// <summary>
+        ///     Whether to automatically set item-prefixes when toggling the flashlight.
+        /// </summary>
+        /// <remarks>
+        ///     Flashlights should probably be using explicit unshaded sprite, in-hand and clothing layers, this is
+        ///     mostly here for backwards compatibility.
+        /// </remarks>
+        [DataField("addPrefix")]
+        public bool AddPrefix = false;
+
+        /// <summary>
+        ///     Sprite layer that will have it's visibility toggled when this item is toggled.
+        /// </summary>
+        [DataField("layer")]
+        public string Layer = "light";
+
+        /// <summary>
+        ///     Layers to add to the sprite of the player that is holding this entity.
+        /// </summary>
+        [DataField("inhandVisuals")]
+        public Dictionary<HandLocation, List<PrototypeLayerData>> InhandVisuals = new();
+
+        /// <summary>
+        ///     Layers to add to the sprite of the player that is wearing this entity.
+        /// </summary>
+        [DataField("clothingVisuals")]
+        public readonly Dictionary<string, List<PrototypeLayerData>> ClothingVisuals = new();
```

Comment:
theres just a shit ton of boilerplate for this tbh. not opposed to these being in since its already coded but for the future this probably needs an easier way to do it

Reply:
The `AddPrefix` field is just for backwards compatibility, because some flashlights haven't been split into layers yet and still just replace the item's in-hand sprite with another. So that part is temporary boilerplate.

Otherwise: I guess this could all be replaced with something like the generic enum visualizer, where some visualizer system automatically handles toggling a combination of sprite+clothing+inhand layers whenever some appearance data is toggled. That way it'd basically just require yaml + some minimal c# (setting appearance data).

Reply:
Ill work on this after this is merged cause RGB & in-hand coloration is currently still broken without this.